src/main/java/org/utplsql/api/outputBuffer/OutputBufferProvider.java

```diff
@@ -42,26 +42,34 @@ public static OutputBuffer getCompatibleOutputBuffer(Version databaseVersion, Re
 
     private static boolean hasOutput( Reporter reporter, OracleConnection oraConn ) throws SQLException {
 
-        String sql = "select is_output_reporter " +
-                " from table(ut_runner.get_reporters_list)" +
-                " where ? = substr(reporter_object_name, length(reporter_object_name)-?+1)";
-        try ( PreparedStatement stmt = oraConn.prepareStatement(sql)) {
+        String reporterName = reporter.getTypeName();
+        if ( !reporterName.matches("^[a-zA-Z0-9_]+$"))
+            throw new IllegalArgumentException(String.format("Reporter-Name %s is not valid", reporterName));
+
+        String sql =
+                "declare " +
+                "   l_result int;" +
+                "begin " +
+                "   begin " +
+                "       execute immediate '" +
+                "       begin " +
+                "           :x := case ' || ? || '() is of (ut_output_reporter_base) when true then 1 else 0 end;" +
```

Comment:
I would keep `dbms_assert.valid_sql_name` call.
It's a build in function that checks object name valid and that it is visible to user.
It is far safer than regex check.
You can still have both